packages/atomic/src/utils/ripple.tsx

```diff
@@ -0,0 +1,37 @@
+interface RippleOptions {
+  color: string;
+  /**
+   * When the ripple's parent is not the event's origin
+   */
+  parent?: Element;
+}
+
+const RIPPLE_CLASS = 'ripple';
+
+function setPositionRelativeIfStatic(element: Element) {
+  if (getComputedStyle(element).position === 'static') {
+    element.classList.add('relative');
+  }
+}
+
+export function createRipple(event: MouseEvent, options: RippleOptions) {
+  const button = options.parent ?? (event.currentTarget as Element);
+  const existingRipple = button.getElementsByClassName(RIPPLE_CLASS)[0];
+  existingRipple && existingRipple.remove();
+
+  button.classList.add('overflow-hidden');
+  setPositionRelativeIfStatic(button);
+  Array.from(button.children).forEach(setPositionRelativeIfStatic);
+
+  const ripple = document.createElement('span');
+  ripple.classList.add(RIPPLE_CLASS, `bg-${options.color}`);
```

Comment:
I would prefer not to use tailwind classes here, since this may be re-used in result templates.